lib/Screens/AuthRegistration/body.dart

```diff
@@ -8,126 +9,204 @@ class Body extends StatefulWidget {
-          child: TextField(
-            obscureText: shouldInputBeObscured,
-            keyboardType: inputType,
-            decoration: InputDecoration(
-              contentPadding: EdgeInsets.only(top: 4.5),
-              hintText: inputHint,
-              hintStyle: hintTextStyle,
-              border: InputBorder.none,
-              prefixIcon: Icon(
-                prefixIcon,
+  GlobalKey<FormState> formKey = GlobalKey();
+  String? password = '';
+  String? passwordConfirmation = '';
+
+  Widget _buildNameInputFrame() => Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Text(
+            'Full Name',
+            style: labelTextStyle,
+          ),
+          SizedBox(height: 5),
+          Container(
+            child: TextFormField(
+              autovalidateMode: AutovalidateMode.onUserInteraction,
+              validator: _getValidationLogic(InputType.fullName),
+              keyboardType: TextInputType.name,
+              decoration: getInputDecoration(InputType.fullName),
+              style: TextStyle(
                 color: Colors.white,
               ),
             ),
-            style: TextStyle(
-              color: Colors.white,
+          )
+        ],
+      );
+
+  Widget _buildPhoneInputFrame() => Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Text(
+            'Phone No',
+            style: labelTextStyle,
+          ),
+          SizedBox(height: 5),
+          Container(
+            child: TextFormField(
+              autovalidateMode: AutovalidateMode.onUserInteraction,
+              validator: _getValidationLogic(InputType.phoneNo),
+              keyboardType: TextInputType.phone,
+              decoration: getInputDecoration(InputType.phoneNo),
+              style: TextStyle(
+                color: Colors.white,
+              ),
             ),
+          )
+        ],
+      );
+
+  Widget _buildMailInputFrame() => Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Text(
+            'Email',
+            style: labelTextStyle,
           ),
-        )
-      ],
-    );
+          SizedBox(height: 5),
+          Container(
+            child: TextFormField(
+              autovalidateMode: AutovalidateMode.onUserInteraction,
+              validator: _getValidationLogic(InputType.email),
+              keyboardType: TextInputType.emailAddress,
+              decoration: getInputDecoration(InputType.email),
+              style: TextStyle(
+                color: Colors.white,
+              ),
+            ),
+          )
+        ],
+      );
+
+  Widget _buildPasswordInputFrame() => Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Text(
+            'Password',
+            style: labelTextStyle,
+          ),
+          SizedBox(height: 5),
+          Container(
+            child: TextFormField(
```

Comment:
In this case you could use a `TextEditingController` in stateful widget. I'm not saying your approach is wrong, but a bit more complicated :)

```
final TextEditingController _nameController = TextEditingController();

@override
  void dispose() {
    _nameController.dispose();
    super.dispose();
  }

And inside of `build` method:

TextFormField(
    controller: _nameController,
    decoration: InputDecoration(
      border: OutlineInputBorder(
        borderRadius: BorderRadius.all(
          Radius.circular(12),
        ),
      ),
      labelText: 'Name label',
      hintText: 'Name hint',
    ),
    validator: (value) {
      if (value == null || value.isEmpty) {
        return 'Please enter the name';
      }
      return null;
    },
  ),```

Reply:
Refactoring done : `TextEditingController` instead of "GlobalKey with Form widget"